allennlp/data/iterators/same_lang_iterator.py

```diff
@@ -0,0 +1,46 @@
+from collections import deque, defaultdict
+from typing import Iterable, Deque
+import logging
+import random
+
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.data_iterator import DataIterator
+from allennlp.data.dataset import Batch
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+def split_by_lang(instance_list):
+    insts_by_lang = defaultdict(lambda: [])
+    for inst in instance_list:
+        inst_lang = inst.fields['metadata'].metadata['lang']
+        insts_by_lang[inst_lang].append(inst)
+
+    return iter(insts_by_lang.values())
+
+@DataIterator.register("same_lang")
+class SameLangIterator(DataIterator):
+    """
+
+    Splits batches into batches containing the same language.
```

Comment:
You should say here how the code expects to figure out which instances are the same language (by looking for a `lang` key in a `metadata` field), and say in the docstring that this is intended for multi-lingual applications.